<FE> client 폴더 빌드 에러 발생

## 문제 상황

SignalingClient 시그니처 수정으로 인한 Demo 컴포넌트 빌드 에러

## 스크린샷

```bash
> client@0.0.0 build
> tsc -b && vite build

src/components/Demo.tsx:49:44 - error TS2345: Argument of type 'Map<string | undefined, MediaStream>' is not assignable to parameter of type 'Map<string, WebRTCData>'.
  Type 'string | undefined' is not assignable to type 'string'.
    Type 'undefined' is not assignable to type 'string'.

49       signalingClient(localStream.current, remoteStreamMap.current);
                                              ~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.

npm error Lifecycle script `build` failed with error:
npm error code 2
npm error path /Users/gekim/temp/client
npm error workspace client@0.0.0
npm error location /Users/gekim/temp/client
npm error command failed
npm error command sh -c tsc -b && vite build
```

--- a/client/src/components/Demo.tsx
+++ b/client/src/components/Demo.tsx
@@ -1,15 +1,22 @@
 import { useEffect, useRef, useState } from 'react';
-import signalingClient from '@/socket/signalingClient';
+import signalingClient from '@socket/signalingClient';
+
+interface WebRTCData {
+  peerConnection: RTCPeerConnection;
+  remoteStream: MediaStream;
+  dataChannel: RTCDataChannel;
+  nickName: string;
+}
 
 function Demo() {
   const localVideoRef = useRef<HTMLVideoElement>(null);
-  const remoteStreamMap = useRef<Map<string | undefined, MediaStream>>(new Map());
-  const localStream = useRef<MediaStream | null>(null);
+  const webRTCMap = useRef(new Map<string, WebRTCData>());
+  const localStreamRef = useRef(new MediaStream());
   const [, forceUpdate] = useState({});
 
   const openUserMedia = async () => {
     if (localVideoRef.current) {
-      localVideoRef.current.srcObject = localStream.current;
+      localVideoRef.current.srcObject = localStreamRef.current;
     }
   };
 
@@ -21,7 +28,7 @@ function Demo() {
 
   useEffect(() => {
     const initStream = async () => {
-      localStream.current = await navigator.mediaDevices.getUserMedia({
+      localStreamRef.current = await navigator.mediaDevices.getUserMedia({
         video: true,
         audio: false,
       });
@@ -44,9 +51,9 @@ function Demo() {
         return result;
       };
 
-      remoteStreamMap.current = observableMap;
+      webRTCMap.current = observableMap;
 
-      signalingClient(localStream.current, remoteStreamMap.current);
+      signalingClient(localStreamRef.current, webRTCMap.current);
     };
 
     initStream();
@@ -88,7 +95,7 @@ function Demo() {
             <video width={480} height={360} ref={localVideoRef} autoPlay playsInline />
           </div>
 
-          {[...remoteStreamMap.current].map(([id, stream]) => (
+          {[...webRTCMap.current].map(([id, { remoteStream }]) => (
             <div
               key={String(id)}
               style={{ border: '1px solid black', width: '480px', height: '360px' }}
@@ -96,7 +103,7 @@ function Demo() {
               <video
                 ref={(element) => {
                   if (element) {
-                    element.srcObject = stream;
+                    element.srcObject = remoteStream;
                   }
                 }}
                 autoPlay

